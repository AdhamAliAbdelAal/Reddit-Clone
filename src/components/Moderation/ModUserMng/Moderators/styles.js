import { Box } from '@mui/material';
import { styled } from '@mui/material/styles';

export const QueueBox = styled(Box)(() => ({
  width: 'auto',
  margin: '0 24px 0 24px',
  paddingTop: 24,

}));

export const QueueText = styled(Box)(() => ({
  display: 'flex',
  alignItems: 'center',
  marginBottom: 20,
}));

export const ControlBar = styled(Box)(() => ({
  display: 'flex',
  justifyContent: 'flex-end',
  alignItems: 'center',
  backgroundColor: '#edeff1',
  height: '48px',
  padding: '0 24px',
}));

export const UserContainer = styled(Box)(() => ({
  marginBottom: '36px',
}));
