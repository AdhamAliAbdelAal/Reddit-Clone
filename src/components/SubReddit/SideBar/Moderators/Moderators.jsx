import MailOutlineIcon from '@mui/icons-material/MailOutline'; import {
  AboutCountainer, AboutString, CustomLink, Button, Container, Name, Data, ViewAll, ViewAllContainer,
} from './style';
/**
 * Moderators section in sidebar
 * @component
 * @return {React.Component} - Moderators section in sidebar
 */
function Moderators(props) {
  const { moderatoesName } = props;
  return (
    <>
      <AboutCountainer>
        <AboutString>
          Moderators
        </AboutString>
      </AboutCountainer>
      <CustomLink>
        <Button variant="outlined" padding="4px" fontSize={15} fontWeight="bold" sx={{ m: 1.8 }}>
          <MailOutlineIcon sx={{ margin: '2px' }} />
          {' '}
          Message the Mods
        </Button>
      </CustomLink>
      { moderatoesName?.map((entity, index) => (
        <Container key={`${index + 0}`}>
          <Name>
            u/
            {entity.name}
          </Name>
          <Data>
            {entity.browsIt}
            {' '}
            browse it
          </Data>
        </Container>
      ))}
      <ViewAllContainer>
        <ViewAll>VIEW ALL MODERATORS</ViewAll>
      </ViewAllContainer>
    </>
  );
}

export default Moderators;
