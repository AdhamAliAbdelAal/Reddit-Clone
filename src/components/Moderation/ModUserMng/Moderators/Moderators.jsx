/* eslint-disable import/no-cycle */
/* eslint-disable react/jsx-no-constructed-context-values */
import { Typography } from '@mui/material';
import ErrorOutlineOutlinedIcon from '@mui/icons-material/ErrorOutlineOutlined';
import * as React from 'react';
import {
  QueueBox, QueueText, ControlBar, InstanceContainer,
} from '../styles';
import { ModMainPage } from '../../ModerationMainPage/styles';
import LeavePopUp from './LeavePopUp/LeavePopUp';
import { RedditButton } from '../../styles';
// import NonEmptyModerator from './NonEmptyModerator/NonEmptyModerator';
import Invitation from './InvitationPopUp/InvitationPopUp';
import SearchBar from '../SearchBar/SearchBar';
import ModeratorsList from './NonEmptyModerator/ModeratorsList/ModeratorsList';

export const LeaveContext = React.createContext();
export const InvitationContext = React.createContext();

function Moderator(props) {
  const { subReddit } = props;

  const [openLeave, setOpenLeave] = React.useState(false);
  const handleClickOpenLeave = () => { setOpenLeave(true); };
  const handleClickCloseLeave = () => { setOpenLeave(false); };

  const [openInvitation, setOpenInvitation] = React.useState(false);
  const handleClickOpenInvitation = () => { setOpenInvitation(true); };
  const handleClickCloseInvitation = () => { setOpenInvitation(false); };

  return (
    <ModMainPage>
      <LeaveContext.Provider value={{
        openLeave, handleClickCloseLeave,
      }}
      >
        <LeavePopUp />
      </LeaveContext.Provider>
      <InvitationContext.Provider value={{
        openInvitation, handleClickCloseInvitation,
      }}
      >
        <Invitation />
      </InvitationContext.Provider>
      <ControlBar>
        <RedditButton
          fontSize="14px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="outlined"
          sx={{ marginRight: '12px' }}
          onClick={handleClickOpenLeave}
        >
          Leave as mod
        </RedditButton>
        <RedditButton
          fontSize="14px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="contained"
          onClick={handleClickOpenInvitation}
        >
          Invite user as mode
        </RedditButton>
      </ControlBar>
      <QueueBox>
        <QueueText>
          <Typography variant="h6">
            Moderators of r/
            {subReddit}
          </Typography>
          <ErrorOutlineOutlinedIcon color="primary" />
        </QueueText>
      </QueueBox>
      <SearchBar />
      <InstanceContainer>
        <ModeratorsList />
        {/* <NonEmptyModerator type="0" />
      </InstanceContainer>
      <InstanceContainer>
        <Typography
          color="#1C1C1C"
          padding="0px 0px 8px 24px"
          fontWeight={600}
          fontSize="14px"
        >
          You can edit these moderators
        </Typography>
        <NonEmptyModerator type="1" />
      </InstanceContainer>
      <InstanceContainer>
        <Typography
          color="#1C1C1C"
          padding="0px 0px 8px 24px"
          fontWeight={600}
          fontSize="14px"
        >
          Invited moderators
        </Typography>
        <NonEmptyModerator type="2" /> */}
      </InstanceContainer>
    </ModMainPage>
  );
}

export default Moderator;
