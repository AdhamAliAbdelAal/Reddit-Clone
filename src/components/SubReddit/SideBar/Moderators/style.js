import { styled } from '@mui/material/styles';
import { ButtonBase } from '@mui/material';
import { Link } from 'react-router-dom';

export const AboutCountainer = styled('div')({
  fontSize: 10,
  fontWeight: 700,
  letterSpacing: 0.5,
  backgroundColor: '#ffffff',
  borderRadius: '3px 3px 0 0',
  color: '#1a1a1b',
  display: 'flex',
  fill: '#1a1a1b',
  textTransform: 'uppercase',
  margin: 0,

});

export const AboutString = styled('h2')({
  padding: '12px 0 0',
  margin: 0,
  fontSize: 14,
  fontWeight: 700,
  textTransform: 'none',
});

export const CustomLink = styled(Link)(({ theme }) => ({
  textDecoration: 'none',
  display: 'flex',
  flexDirection: 'column',
  color: theme.palette.primary.main,
  '&:hover': {
    textDecoration: 'none',
  },
}));

export const Button = styled(ButtonBase)(({
  fontSize, fontWeight, backgroundColor,
}) => ({
  boxShadow: 'none',
  backgroundColor,
  fontSize,
  borderRadius: 20,
  textTransform: 'initial',
  minWidth: 'auto',
  fontWeight,
  alignItems: 'center',
  display: 'flex',
  marginBottom: 20,
  justifyContent: 'center',
  padding: '4px 16px',
  textalign: 'center',
  border: '1px solid #0079d3',
  height: 30,
}));

export const Container = styled('div')(() => ({
  fontSize: 12,
  fontWeight: 500,
  display: 'flex',
  marginBottom: 16,
  width: '100%',

}));
export const Name = styled(Link)(() => ({
  cursor: 'pointer',
  color: '#0079d3',
  display: 'inline-block',
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'none',
  },
}));
export const Data = styled('div')(() => ({
  backgroundColor: '#EDEFF1',
  color: '#1A1A1B',
  fontSize: 12,
  fontWeight: 500,
  borderRadius: 2,
  display: 'inline-block',
  marginRight: 5,
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  verticalAlign: 'text-bottom',
  whiteSpace: 'pre',
  wordBreak: 'normal',
  padding: '0px 4px',
  marginLeft: 4,
}));
export const ViewAllContainer = styled('div')(() => ({
  display: 'flex',
  overflow: 'hidden',
  padding: '0 12px 12px',
}));
export const ViewAll = styled(Link)(() => ({
  fontSize: 12,
  fontWeight: 700,
  letterSpacing: '.5px',
  textTransform: 'uppercase',
  color: '#0079d3',
  marginLeft: 'auto',
  cursor: 'pointer',
}));
