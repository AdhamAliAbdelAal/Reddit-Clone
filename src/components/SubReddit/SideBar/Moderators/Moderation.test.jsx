import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Moderators from './Moderators';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<Moderators />).toJSON();
  expect(tree).toMatchSnapshot();
});
