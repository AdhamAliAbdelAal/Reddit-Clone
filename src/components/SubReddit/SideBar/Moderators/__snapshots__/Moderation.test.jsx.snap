// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
Array [
  <div
    className="css-1f2ln3t"
  >
    <h2
      className="css-dgde1h"
    >
      Moderators
    </h2>
  </div>,
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-jf1sgt-MuiTypography-root-MuiLink-root"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <button
      className="MuiButtonBase-root css-azyh4b-MuiButtonBase-root"
      disabled={false}
      fontSize={15}
      fontWeight="bold"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="4px"
      tabIndex={0}
      type="button"
      variant="outlined"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ubwraf-MuiSvgIcon-root"
        data-testid="MailOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
        />
      </svg>
       
      Message the Mods
    </button>
  </a>,
  <div
    className="css-e3tcub"
  >
    <div
      className="css-qq790j"
    >
      VIEW ALL MODERATORS
    </div>
  </div>,
]
`;
